### ISSUE TITLE
Feature importance for random forests

### ISSUE THREAD

**Issue by mjmckp** [dotnet/machinelearning#210]
When training a random forest, the scikit-learn package returns the feature importances (see http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html), where the feature importance metric is the mean decrease impurity (as described here http://blog.datadive.net/selecting-good-features-part-iii-random-forests/).  Can `ML.Net` also provide this information?

**Comment by codemzs** [comment]
Hi @mjmckp , Yes, this information is [maintained in the runtime](https://github.com/dotnet/machinelearning/blob/c023727b76970ab913ec1ce38276508835c17bcf/src/Microsoft.ML.FastTree/TreeEnsemble/RegressionTree.cs#L1346) but there isn't a high level API at the moment that exposes feature importance. We are working on exposing model summary via high level API that should contain feature importance.

**Comment by zeahmed** [comment]
Looking at the new API proposal #371, this feature will be automatically available. 

For example, in the [proposed example](https://github.com/TomFinley/machinelearning/blob/deedfcde1c8f716fa725ca8a75497513c493e629/test/Microsoft.ML.Tests/Scenarios/DirectInstantiation.cs#L172) look at [this Line](https://github.com/TomFinley/machinelearning/blob/deedfcde1c8f716fa725ca8a75497513c493e629/test/Microsoft.ML.Tests/Scenarios/DirectInstantiation.cs#L221) where predictor is created. The predictor contains several methods to save models/metadata and `SaveSummary` is one of them which emits feature importance where applicable.


Also if the predictor implements `ICanGetSummaryInKeyValuePairs` then feature importance can be obtain as a List of KeyValuePairs. For example, see [here](https://github.com/dotnet/machinelearning/blob/8b01fc5a8875fd8543777d8ec4d0bf370bc590f5/src/Microsoft.ML.FastTree/FastTree.cs#L2778) for reference.

**Comment by shauheen** [comment]
Closing this issue as #371 would address the main problem.
